force-app/main/default/classes/DMLOperations.cls

```diff
@@ -159,36 +212,69 @@ public with sharing class DMLOperations {
 	 * @return The Account that was upserted.
 	 */
 	public static Account upsertAccount(String accountName) {
-		// Query for the Account
-		List<Account> accounts = [SELECT Id FROM Account WHERE Name = :accountName];
-
+		//create new instance of acct obj
+        Account newAcct = new Account();
+        // Query for the Account
+		List<Account> accounts = [SELECT Id FROM Account WHERE Name = :accountName LIMIT 1];
 		// if the Account exists, update the description
+        if (!accounts.isEmpty()){
+             newAcct = accounts.get(0);
+             newAcct.Description = 'Updated Account';
+             upsert newAcct;
+             return newAcct;  
+        
+        } else {
+            // otherwise, create a new Account and set the description
+            newAcct.Description = 'New Account';
+            newAcct.Name = accountName;
+		    // Upsert the Account
+            upsert newAcct;
+            return newAcct;
```

Comment:
Several of these lines are the same in both conditions of your if else. What about moving them to after the end of the else, that way you only have to write them once.